ibis/sql/tests/test_compiler.py

```diff
@@ -1459,25 +1459,24 @@ def test_subquery_in_union(self):
         expr = join1.union(join2)
         result = to_sql(expr)
         expected = """\
-(WITH t0 AS (
+WITH t0 AS (
   SELECT `a`, `g`, sum(`f`) AS `metric`
   FROM alltypes
   GROUP BY 1, 2
+),
+t1 AS (
+  SELECT t0.*
+  FROM t0
+    INNER JOIN t0 t3
+      ON t0.`g` = t3.`g`
 )
-SELECT t0.*
-FROM t0
-  INNER JOIN t0 t1
-    ON t0.`g` = t1.`g`)
+SELECT *
+FROM t1
 UNION ALL
-(WITH t0 AS (
-  SELECT `a`, `g`, sum(`f`) AS `metric`
-  FROM alltypes
-  GROUP BY 1, 2
-)
 SELECT t0.*
 FROM t0
-  INNER JOIN t0 t1
-    ON t0.`g` = t1.`g`)"""
+  INNER JOIN t0 t3
+    ON t0.`g` = t3.`g`"""
```

Comment:
This isn't factored out as `t1` because `join1` and `join2` are not the same ibis expression.